experiment/test/cortex/experiment/tensorboard_test.clj

```diff
@@ -0,0 +1,138 @@
+                           :test-accuracy (get-metric met/accuracy test-ds labels)
+                           :train-precision (get-metric met/precision train-ds predicted-train-ds-labels)
+                           :test-precision (get-metric met/precision test-ds labels)
+                           :train-recall (get-metric met/recall train-ds predicted-train-ds-labels)
+                           :test-recall (get-metric met/recall test-ds labels)))
+        _ (go (>! metrics-chan (bcast-msg)))
+        current-best-loss (if-let [best-loss (get old-network :cv-loss)]
+                            ;; TODO: Is there a bug here? What if the best-loss isn't sequential?
+                            (when (sequential? best-loss)
+                              (apply + (map :value best-loss))))
+        best-network? (or (nil? current-best-loss)
+                          (< (double loss-val)
+                             (double current-best-loss)))]
+    ;(println (format "Loss for epoch %s: %s" (get new-network :epoch-count) loss-val))
+    {:best-network? best-network?
+     :network (assoc new-network :cv-loss loss-fn)}))
+
+(defn- spit-events
+  [file-path metrics]
+  (let [ev (mapv eio/make-event
+                 (mapv (partial str "metrics/") (mapv name (keys metrics)))
+                 (vals metrics))]
+    (eio/append-events file-path ev)))
+
+(defn- start-tensorboard-broadcast
+  [file-path channel num-msgs]
+  (do
+    (eio/create-event-stream file-path)
+    (go (dotimes [n num-msgs]
+          (let [[v _] (alts! [channel])]
+            (spit-events file-path v))))))
+
+(defn- train-and-eval
+  [train-ds test-ds network-description log-file]
+  (let [epoch-count 20
+        metrics-chan (chan)
+        event-writer-chan (start-tensorboard-broadcast log-file metrics-chan epoch-count)
+        _ (train/train-n network-description train-ds test-ds
+                         :batch-size 50 :epoch-count epoch-count
+                         :test-fn (partial log-eval-metrics metrics-chan))]
+    ;;wait for the listener to finish reading epoch-count events
+    (<!! event-writer-chan)))
+
+(deftest test-listener
+  (let [ds (shuffle (default-dataset))
+          log-path "/tmp/tflogs/"
+          ds-count (count ds)
+          train-ds (take (int (* 0.9 ds-count)) ds)
+          test-ds (drop (int (* 0.9 ds-count)) ds)
+          train-fn (partial train-and-eval train-ds test-ds)]
+      (mapv train-fn [linear-with-batch-norm
+                      linear-without-batch-norm relu-with-batch-norm
+                      tanh-with-batch-norm]
+            (mapv #(let [path (str log-path % "/tfevents." % ".out")]
+                     (io/make-parents path)
+                     path)
+                  ["linear"
+                   "linear_nobatchnorm" "relu" "tanh"]))
+      (is (.exists (io/file (str log-path "/linear/tfevents.linear.out"))))))
```

Comment:
This is certainly interesting, and runs to completion without errors for me.

It's definitely a great example of integrating your library into cortex, but I'm not sure how this fits into our visualization strategy more broadly.

Can you include some instructions (or a pointer to some instructions) on how to go from the contents of `/tmp/tflogs` to the attractive screenshot you posted? Would such instructions belong in the README of the `org.shark8me/tfevent-sink` repo?